app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -127,10 +127,12 @@ class BrowserFragment : IWebViewLifecycleFragment() {
                         }
                         NavigationEvent.VAL_UNCHECKED -> {
                             url.toUri()?.let {
-                                if (BundledTilesManager.getInstance(context).unpinSite(context, it)
-                                        || CustomTilesManager.getInstance(context).unpinSite(context, url)) {
-                                    showCenteredTopToast(context, R.string.notification_unpinned_site, 0, TOAST_Y_OFFSET)
+                                if (BundledTilesManager.getInstance(context).unpinSite(context, it)) {
+                                    browserOverlay.removePinnedSiteFromTiles(it.toString())
+                                } else if (CustomTilesManager.getInstance(context).unpinSite(context, url)) {
+                                    browserOverlay.removePinnedSiteFromTiles(url)
                                 }
+                                showCenteredTopToast(context, R.string.notification_unpinned_site, 0, TOAST_Y_OFFSET)
```

Comment:
This isn't correct, the original code shows a toast if one of these cases is true.